crates/python_utilities/src/ast_printer.rs

```diff
@@ -164,6 +175,148 @@ fn emit_body<'a>(body: &'a [Stmt], str_buffer: &mut WritingBuffer<'a>) {
+ *         the function name.
+ */
+trait CustomDisplay {
+    fn custom_fmt(&self, str_buffer: &mut WritingBuffer, defer: bool) -> String;
+}
+
+impl CustomDisplay for ast::Expr {
+    fn custom_fmt(&self, str_buffer: &mut WritingBuffer, defer: bool) -> String {
+        fn push(str_buffer: &mut WritingBuffer, defer: bool, s: String) -> String {
+            if !defer {
+                str_buffer.push(Cow::Owned(s.clone()));
+            }
+            s
+        }
+        fn push_inline_list(
+            str_buffer: &mut WritingBuffer,
+            defer: bool,
+            list: &Vec<ast::Expr>,
+        ) -> String {
+            str_buffer.push(Cow::Owned("[".to_string()));
+            let mut idx: usize = 0;
+            for elem in list.iter() {
+                elem.custom_fmt(str_buffer, defer);
+                if idx < list.len() - 1 {
+                    str_buffer.push(Cow::Owned(", ".to_string()))
+                }
+                idx += 1;
+            }
+            str_buffer.push(Cow::Owned("]".to_string()));
+            "".to_string()
+        }
+        fn push_multi_line_list(
+            str_buffer: &mut WritingBuffer,
+            defer: bool,
+            list: &Vec<ast::Expr>,
+        ) -> String {
+            str_buffer.push(Cow::Owned("[".to_string()));
+            str_buffer.finish_line();
+            str_buffer.indent();
+            for elem in list.iter() {
+                elem.custom_fmt(str_buffer, defer);
+                str_buffer.push(Cow::Owned(",".to_string()));
+                str_buffer.finish_line();
+            }
+            str_buffer.deindent();
+            str_buffer.push(Cow::Owned("]".to_string()));
+            "".to_string()
+        }
+
+        match self {
+            // This uses double-quotation for String literals
+            ast::Expr::Constant(ast::ExprConstant { value, .. }) => match value {
+                ast::Constant::Str(str) => push(str_buffer, defer, format!("\"{}\"", str)),
+                _ => push(str_buffer, defer, format!("{}", self)),
+            },
+            ast::Expr::Call(ast::ExprCall {
+                func,
+                args,
+                keywords,
+                ..
+            }) => {
+                let func_expr: &ast::Expr = &*func;
+                let name = func_expr.custom_fmt(str_buffer, true);
+                if name == "load" {
+                    str_buffer.push(Cow::Owned(name));
+                    str_buffer.push(Cow::Owned("(".to_string()));
+                    let mut idx: usize = 0;
+                    for arg in args.iter() {
+                        arg.custom_fmt(str_buffer, defer);
+                        if idx < args.len() - 1 {
+                            str_buffer.push(Cow::Owned(", ".to_string()))
+                        }
+                        idx += 1;
+                    }
+                    idx = 0;
+                    for kw in keywords.iter() {
```

Comment:
again implementing `enumerate()`

Reply:
Addressed in b8ddcf6079fe659b1cd1c94fd82d5da6e6238c2f